processing/src/main/java/org/apache/druid/segment/serde/ColumnPartSerde.java

```diff
@@ -55,5 +56,18 @@ public interface ColumnPartSerde
   interface Deserializer
   {
     void read(ByteBuffer buffer, ColumnBuilder builder, ColumnConfig columnConfig);
+
+    default void readProjection(
+        ByteBuffer buffer,
+        ColumnBuilder builder,
+        ColumnConfig columnConfig,
+        ColumnHolder parent
```

Comment:
this should be marked nullable and null-checked in the implementations.

Reply:
i think that ideally this shouldn't be called if the parent column doesn't exist since the use case here is basically about attaching stuff from the base table column to the projection column, and we can just load the projection column directly because it has all of its own stuff if it doesn't exist on the base table

Reply:
ok. It was marked nullable in the ColumnDescriptor#readProjection so maybe it shouldn't be nullable there.